src/test/java/net/sf/jabref/importer/fileformat/BibtexParserTest.java

```diff
@@ -1486,4 +1487,43 @@ public void parseReturnsEntriesInSameOrder() throws IOException {
 
         assertEquals(expected, result.getDatabase().getEntries());
     }
+
+    @Test
+    public void parsePrecedingComment() throws IOException {
+        // @formatter:off
+        String bibtexEntry = "% Some random comment that should stay here" + Globals.NEWLINE +
```

Comment:
This does not make a difference. Any non-whitespace text (except for the `ENCODING_PREFIX`) is treated as a comment. In a sense, the percentage sign here is a complete coincidence.